tests/integration/clientv3/watch_test.go

```diff
@@ -74,6 +74,12 @@ func testWatchMultiWatcher(t *testing.T, wctx *watchctx) {
 	keys := []string{"foo", "bar", "baz"}
 
 	donec := make(chan struct{})
+	// wait for watcher shutdown
+	defer func() {
+		for i := 0; i < len(keys)+1; i++ {
+			<-donec
+		}
+	}()
```

Comment:
The goroutines may be still running when the test case finishes/returns. So it may also cause another goroutine leak.  I would suggest to keep it as it's for now.

Reply:
It should be OK, the test case will not return until the `defer` function finishes